### DISCUSSION TITLE
TSKPaintBox

### DISCUSSION THREAD

**Discussion by RobertoColpani** [skia4delphi/skia4delphi#236] [Q&A]
How can I print a rotate text of 45°?

**Comment by viniciusfbb** [comment]
The ISkCanvas has internal states of the drawing (like the matrix for the drawings), which you can change once and all drawings onwards will be applied to the new state. ISkCanvas.Rotate() works this way. Then you can save the state to rollback the state change you made after drawing the rotated text. You will understand better with the code:

```delphi
ACanvas.Save;
try
   ACanvas.Rotate(45);
   // Draw my text
finally
   ACanvas.Restore;
end;
```
